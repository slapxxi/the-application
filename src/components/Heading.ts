import * as _ from 'lodash';
import styled from 'styled-components';

const Heading = styled('h1')`
  font-family: Roboto Slab, serif;
  font-size: 34px;
  line-height: 1.1764;
  color: ${(p) => _.get(p, 'theme.colors.heading')};
`;

export default Heading;
