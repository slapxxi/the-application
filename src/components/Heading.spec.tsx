import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import Heading from './Heading';

const theme = {
  colors: {
    heading: 'black',
  },
};

it('renders', () => {
  const component = renderer.create(<Heading />).toJSON();
  expect(component).toMatchSnapshot();
});

it('accepts `theme` prop', () => {
  const component = renderer
    .create(<Heading theme={theme} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it('does not throw if `theme` is malformed', () => {
  expect(() =>
    render(<Heading theme={{}}>text</Heading>),
  ).not.toThrow();
});
