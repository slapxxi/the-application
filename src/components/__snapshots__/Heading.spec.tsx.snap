// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts \`theme\` prop 1`] = `
.c0 {
  font-family: Roboto Slab,serif;
  font-size: 34px;
  line-height: 1.1764;
  color: black;
}

<h1
  className="c0"
/>
`;

exports[`renders 1`] = `
.c0 {
  font-family: Roboto Slab,serif;
  font-size: 34px;
  line-height: 1.1764;
}

<h1
  className="c0"
/>
`;
